modules/internal/String.chpl

```diff
@@ -218,62 +218,70 @@ module String {
   }
 
   /*
-     A value of type :record:`codePointIndex` can be passed to certain
+     A value of type :record:`codepointIndex` can be passed to certain
      `string` functions to indicate that the function should operate
-     with units of code points. See :proc:`~string.this`.
+     with units of codepoints. See :proc:`~string.this`.
 
-     To create or modify a :record:`codePointIndex`, cast it to or from an
+     To create or modify a :record:`codepointIndex`, cast it to or from an
      `int`. For example, For example, the following function returns a string
      containing only the second codepoint of the argument:
 
      .. code-block:: chapel
 
        proc getSecondCodepoint(arg:string) : int {
-         var offsetInCodepoints = 1:codePointIndex;
+         var offsetInCodepoints = 1:codepointIndex;
          return arg[offsetInCodepoints];
        }
 
    */
-  record codePointIndex {
+  record codepointIndex {
     pragma "no doc"
     var _cpindex  : int;
   }
 
-  // Helper routines in support of being able to use ranges of codePointIndex
+  /*
+   Deprecated, use `codepointIndex`.
+  */
+  proc codePointIndex type {
```

Comment:
Normally we `pragma "no doc"` deprecated types/functions

Reply:
I don't think this is true. When this was last discussed, it was determined that we would continue to document deprecated features. This was documented in the [deprecation guidelines](https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst):

> Continue to document and support the deprecated feature for 1 release cycle - The documentation should contain a pointer to the preferred feature - This holds true for modules as well

Reply:
Ah, then I must be doing it wrong...